src/components/delegate/delegate-section.tsx

```diff
@@ -0,0 +1,105 @@
+import { RVR_AUTHORIZER, RVR_TOKEN } from "@/constants/contracts"
+import { cn } from "@/lib/utils"
+import { useAccount, useReadContract, useReadContracts } from "wagmi"
+import { Typography } from "../ui/typography"
+import { WalletAddress } from "../wallet-address"
+import { formatUnits } from "viem"
+import { Skeleton } from "../ui/skeleton"
+import { DelegateForm } from "./delegate-form"
+import { AuthorizeClaimerForm } from "./authorize-claimer-form"
+
+export const DelegateSection = () => {
+  const { address } = useAccount()
+
+  const riverToken = useReadContracts({
+    allowFailure: false,
+    contracts: [
+      {
+        address: RVR_TOKEN.sepolia,
+        abi: RVR_TOKEN.abi,
+        functionName: 'balanceOf',
+        args: address ? [address] : undefined,
+      },
+      {
+        address: RVR_TOKEN.sepolia,
+        abi: RVR_TOKEN.abi,
+        functionName: 'decimals',
+      },
+    ],
+    query: {
+      enabled: !!address,
+    },
+  })
+
+  const delegatee = useReadContract({
+    address: RVR_TOKEN.sepolia,
+    abi: RVR_TOKEN.abi,
+    functionName: 'delegates',
+    args: address ? [address] : undefined,
+  })
+
+  const authorizedClaimer = useReadContract({
```

Comment:
Same here as previous comment. We should maintain state around the network a user is connected to and the contract address, chainId used to run the transaction should all flow downstream from that state.